tests/FSharp.Test.Utilities/InlineFileDataAttribute.fs

```diff
@@ -0,0 +1,160 @@
+/// Takes a file, relative to current test suite's root.
+/// Returns a CompilationUnit with encapsulated source code, error baseline and IL baseline (if any).
+[<NoComparison; NoEquality>]
+type FileInlineData (directory: string, filename: string) =
+    inherit DataAttribute()
+
+    let mutable optimize: BooleanOptions option = None
+    let mutable realsig: BooleanOptions option = None
+
+    static let computeBoolValues opt =
+        match opt with
+        | Some BooleanOptions.True -> [|Some true|]
+        | Some BooleanOptions.False -> [|Some false|]
+        | Some BooleanOptions.Both -> [|Some true; Some false|]
+        | _ -> [|None|]
+
+    static let convertToNullableBool (opt: bool option): obj =
+        match opt with
+        | None -> null
+        | Some opt -> box opt
+
+    member _.Optimize with set v = optimize <- Some v
+
+    member _.Realsig with set v = realsig <- Some v
+
+    override _.GetData _ =
+
+        let getOptions realsig optimize =
+
+            let compilationHelper = CompilationHelper(filename, directory, convertToNullableBool realsig, convertToNullableBool optimize)
+            [| box (compilationHelper) |]
+
+        let results =
+            let rsValues = computeBoolValues realsig
+            let optValues = computeBoolValues optimize
+            [|
+                for r in rsValues do
+                    for o in optValues do
+                        getOptions r o
+            |]
+
+        results
+
+// realsig and optimized are boxed so null = not set, true or false = set
+and CompilationHelper internal (filename: string, directory: string, realsig: obj, optimize: obj) =
+
+    let mutable filename = filename
+    let mutable directory = directory
+    let mutable realsig = realsig
+    let mutable optimize = optimize
+
+    let setRealInternalSignature compilation =
+        match realsig with
+        | null -> compilation
+        | realsig -> compilation |> withRealInternalSignature (unbox realsig)
+
+    let setOptimization compilation =
+        match optimize with
+        | null -> compilation
+        | optimize -> compilation |> withOptimization (unbox optimize)
+
+    static let getBaseline (opt: obj) prefix =
+        match opt with
+        | :? bool as b when b = true -> Some $"{prefix}On"
+        | :? bool as b when b = false -> Some $"{prefix}Off"
+        | _ -> None
+
+    static let combineBaselines realsigBsl optimizeBsl =
+        match realsigBsl, optimizeBsl with
+        | Some rs, Some opt -> Some $"{rs}{opt}"
+        | Some rs, None -> Some $"{rs}"
+        | None, Some opt -> Some $"{opt}"
+        | _ -> None
+
+    new() = CompilationHelper("", "", null, null)
+
+    // Define the implicit conversion operator
+    static member op_Implicit(helper: CompilationHelper) : CompilationUnit =
+        helper.Value ()
+
+    member _.Value(): CompilationUnit =
+        let frame = StackTrace().GetFrame(1) // Get the calling method's frame
```

Comment:
@T-Gro , no.   The Xunit.Sdk.DataAttribute.GetData  override requires a MethodInfo instance. The caller-information attributes return the names of the member, the source file and line numbers.

Here is the Xunit description:  
https://csharp-tokyo.github.io/xUnit-Hands-on/class_xunit_1_1_sdk_1_1_data_attribute.html#ab74ea9efe0a313e19ac0e8d8136a9071

Reply:
We need it, because GetData _ = is how XUnit retrieves the test data instance, and in this case we are using the DirectoryAttribute to create a compiler instance, and so we need to 'pretend' we are being xunit.  It is true that our DirectoryAttribute doesn't actually use it, but we do need an instance to even invoke the method.  And there is a decent probablility that at some point in the future DirectoryAttribute.GetData() will use it, so I didn't want to pass Unchecked.DefaultOf<MethodInfo>.

Reply:
@majocha --- I'm still trying to figure out what we need.  But we can certainly refactor, if necessary once we understand the necessities.